src/govuk/components/pagination/template.njk

```diff
@@ -0,0 +1,62 @@
+{% set type = params.results.type | default("results") -%}
+
+<nav class="govuk-pagination {{- ' ' + params.classes if params.classes}}" id="pagination-label" aria-label="{{type}} navigation">
+
+  <ul class="govuk-pagination__list">
+    {%- if params.previous %}
+    <li class="govuk-pagination__item govuk-pagination__item--prev">
+      <a class="govuk-pagination__link" href="{{ params.previous.href }}">
+        <span class="govuk-pagination__link-title">
+          <svg class="govuk-pagination__icon" xmlns="http://www.w3.org/2000/svg" height="13" width="17">
+            <path d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"></path>
+          </svg>{%- if params.previous.text %}{{ params.previous.text }}{% else %}Previous
+          <span class="govuk-visually-hidden">page</span>
+          {% endif -%}
+        </span>
+        {%- if params.previous.label %}
+        <span class="govuk-pagination__link-divider govuk-visually-hidden">:</span>
+        <span class="govuk-pagination__link-label">{{ params.previous.label }}</span>
+        {% endif -%}
+      </a>
+    </li>
+    {% endif -%}
+
+    {%- for item in params.items %}
+    {%- if item.type == 'elipses' %}
+    <li class="govuk-pagination__item govuk-pagination__item--elipses">&hellip;</li>
+    {% else %}
+    {%- if item.selected %}
+    <li class="govuk-pagination__item govuk-pagination__item--active"><span class="govuk-visually-hidden">Page </span>{{ item.text }}<span class="govuk-visually-hidden"> (current page) </span></li>
+    {% else %}
+    <li class="govuk-pagination__item"><a class="govuk-pagination__link" href="{{ item.href }}"><span class="govuk-visually-hidden">Page </span>{{ item.text }}</a></li>
+    {% endif -%}
+    {% endif -%}
+    {% endfor -%}
+
+    {%- if params.next %}
+    <li class="govuk-pagination__item govuk-pagination__item--next">
+      <a class="govuk-pagination__link" href="{{ params.next.href }}">
+        <span class="govuk-pagination__link-title">
+          <svg class="govuk-pagination__icon" xmlns="http://www.w3.org/2000/svg" height="13" width="17">
+            <path d="m10.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z"></path>
+          </svg>
+          {%- if params.next.text %}
+          {{ params.next.text }}
+          {% else %}
+          Next <span class="govuk-visually-hidden">page</span>
+          {% endif -%}
+        </span>
+        {%- if params.next.label %}
+        <span class="govuk-pagination__link-divider govuk-visually-hidden">:</span>
+        <span class="govuk-pagination__link-label">{{ params.next.label }}</span>
+        {% endif -%}
+      </a>
+    </li>
+    {% endif -%}
+  </ul>
+
+  {%- if params.results %}
+  <p class="govuk-pagination__results">Showing <b>{{ params.results.from }}</b> to <b>{{ params.results.to }}</b> of <b>{{params.results.count}}</b> {{type}}</p>
```

Comment:
Add some extra classes here, just in case we (or anyone) want to style them differently in future?

```suggestion
  <p class="govuk-pagination__results">Showing <b class="govuk-pagination__results__from">{{ params.results.from }}</b> to <b class="govuk-pagination__results__to">{{ params.results.to }}</b> of <b class="govuk-pagination__results__count">{{params.results.count}}</b> {{type}}</p>
```

Alternatively, maybe the whole line should be passed in to the macro as html (or text)?

```suggestion
  <p class="govuk-pagination__results">{{ params.results.html }}</p>
```

Reply:
I like the idea of custom HTML, since those parameters aren't used for anything else and it would allow for customisation.

If you do keep the embedded parameters though, they should use `<strong>` tags rather than `<b>`

Reply:
Thanks both - again good points. I'll leave this thread here for when we finalise the macro design with the team.